### PR TITLE
simple floodwait fix

### PR DESCRIPTION
**Pull request by ve1x**
When sending a list of requests if one of the requests gets FloodWait then at the next attempt telethon tries to handle just request which got FloodWait and throw away others. Here is a simple fix

### CODE REVIEW

**Comment by ve1x** [comment]
I've got this error calling `iter_participants`

**Comment by ve1x** [comment]
aslo `request_index = 0` is required in FloodWait exception section

File: telethon/client/users.py [hunk 1]
```diff
@@ -30,14 +30,14 @@ class UserMethods(TelegramBaseClient):
...
-
+        results = None
         request_index = 0
         self._last_request = time.time()
         for _ in range(self._request_retries):
             try:
                 future = self._sender.send(request, ordered=ordered)
                 if isinstance(future, list):
-                    results = []
+                    results = [] if not results else results
```
Comment (Lonami) [inline telethon/client/users.py]: `results or []` would've worked. Might as well start with `results = []` above.

**Comment by Lonami** [comment]
The call code is kind of messy to be honest. I should fix that, but right now it's hard to follow the control logic, although I think it's right.
